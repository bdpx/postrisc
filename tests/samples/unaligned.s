.text
.data
data_unaligned:
align 16
    d1	0x00
    d1	0x01
    d1	0x02
    d1	0x03
    d1	0x04
    d1	0x05
    d1	0x06
    d1	0x07
    d1	0x08
    d1	0x09
    d1	0x0a
    d1	0x0b
    d1	0x0c
    d1	0x0d
    d1	0x0e
    d1	0x0f

    d1	0x10
    d1	0x11
    d1	0x12
    d1	0x13
    d1	0x14
    d1	0x15
    d1	0x16
    d1	0x17
    d1	0x18
    d1	0x19
    d1	0x1a
    d1	0x1b
    d1	0x1c
    d1	0x1d
    d1	0x1e
    d1	0x1f

.text
    write	"load/store unaligned"
    alloc	96
    ca.rf	%r17, data_unaligned

    ldz.h	%r3, %r17, 0
    write	"%x16(r3)"
    ldz.h	%r3, %r17, 1
    write	"%x16(r3)"
    ldz.h	%r3, %r17, 2
    write	"%x16(r3)"

    ldz.w	%r3, %r17, 0
    write	"%x32(r3)"
    ldz.w	%r3, %r17, 1
    write	"%x32(r3)"
    ldz.w	%r3, %r17, 2
    write	"%x32(r3)"
    ldz.w	%r3, %r17, 3
    write	"%x32(r3)"
    ldz.w	%r3, %r17, 4
    write	"%x32(r3)"

    ldz.d	%r3, %r17, 0
    write	"%x64(r3)"
    ldz.d	%r3, %r17, 1
    write	"%x64(r3)"
    ldz.d	%r3, %r17, 2
    write	"%x64(r3)"
    ldz.d	%r3, %r17, 3
    write	"%x64(r3)"
    ldz.d	%r3, %r17, 4
    write	"%x64(r3)"
    ldz.d	%r3, %r17, 5
    write	"%x64(r3)"
    ldz.d	%r3, %r17, 6
    write	"%x64(r3)"
    ldz.d	%r3, %r17, 7
    write	"%x64(r3)"
    ldz.d	%r3, %r17, 8
    write	"%x64(r3)"

    ld.q	%r3, %r17, 0
    write	"%x128(r3)"
    ld.q	%r3, %r17, 1
    write	"%x128(r3)"
    ld.q	%r3, %r17, 2
    write	"%x128(r3)"
    ld.q	%r3, %r17, 3
    write	"%x128(r3)"
    ld.q	%r3, %r17, 4
    write	"%x128(r3)"
    ld.q	%r3, %r17, 5
    write	"%x128(r3)"
    ld.q	%r3, %r17, 6
    write	"%x128(r3)"
    ld.q	%r3, %r17, 7
    write	"%x128(r3)"
    ld.q	%r3, %r17, 8
    write	"%x128(r3)"
    ld.q	%r3, %r17, 9
    write	"%x128(r3)"
    ld.q	%r3, %r17, 10
    write	"%x128(r3)"
    ld.q	%r3, %r17, 11
    write	"%x128(r3)"
    ld.q	%r3, %r17, 12
    write	"%x128(r3)"
    ld.q	%r3, %r17, 13
    write	"%x128(r3)"
    ld.q	%r3, %r17, 14
    write	"%x128(r3)"
    ld.q	%r3, %r17, 15
    write	"%x128(r3)"
    ld.q	%r3, %r17, 16
    write	"%x128(r3)"
.end
