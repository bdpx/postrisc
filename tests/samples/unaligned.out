load/store unaligned
0100
0201
0302
03020100
04030201
05040302
06050403
07060504
0706050403020100
0807060504030201
0908070605040302
0a09080706050403
0b0a090807060504
0c0b0a0908070605
0d0c0b0a09080706
0e0d0c0b0a090807
0f0e0d0c0b0a0908
0f0e0d0c0b0a0908_0706050403020100
100f0e0d0c0b0a09_0807060504030201
11100f0e0d0c0b0a_0908070605040302
1211100f0e0d0c0b_0a09080706050403
131211100f0e0d0c_0b0a090807060504
14131211100f0e0d_0c0b0a0908070605
1514131211100f0e_0d0c0b0a09080706
161514131211100f_0e0d0c0b0a090807
1716151413121110_0f0e0d0c0b0a0908
1817161514131211_100f0e0d0c0b0a09
1918171615141312_11100f0e0d0c0b0a
1a19181716151413_1211100f0e0d0c0b
1b1a191817161514_131211100f0e0d0c
1c1b1a1918171615_14131211100f0e0d
1d1c1b1a19181716_1514131211100f0e
1e1d1c1b1a191817_161514131211100f
1f1e1d1c1b1a1918_1716151413121110
